test/integ-new/main_new_test.go

```diff
@@ -0,0 +1,196 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This folder (test/integ-new) will be used to replace
+// folder test/integ after all new integ tests are implemented.
+// For now these tests will be ran by ci-new.yml.
+// If you are making changes to this file, please manully
+// run the ci-new workflow to make sure things works.
+
+// Main entry point for integration tests.
+package integration
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/abcxyz/jvs/pkg/cli"
+	"github.com/google/go-cmp/cmp"
+)
+
+var (
+	// Global integration test config.
+	cfg *config
+
+	// Keys we don't compare in validation result.
+	ignoreKeysMap map[string]struct{}
+)
+
+const (
+	timestampFormat = "2006-01-02 3:04PM UTC"
+	testTTLString   = "30m"
+	testTTLTime     = 30 * time.Minute
+)
+
+func TestMain(m *testing.M) {
+	os.Exit(func() int {
+		ctx := context.Background()
+
+		ignoreKeysMap = map[string]struct{}{
+			"nbf": {},
+			"jti": {},
+		}
+
+		if strings.ToLower(os.Getenv("TEST_INTEGRATION")) != "true" {
+			log.Printf("skipping (not integration)")
+			// Not integration test. Exit.
+			return 0
+		}
+
+		// set up global test config.
+		c, err := newTestConfig(ctx)
+		if err != nil {
+			log.Printf("Failed to parse integration test config: %v", err)
+			return 2
+		}
+		cfg = c
+
+		return m.Run()
+	}())
+}
+
+func TestAPIAndPublicKeyService(t *testing.T) {
+	t.Parallel()
+	now := time.Now()
+	cases := []struct {
+		name                 string
+		isBreakglass         bool
+		justification        string
+		wantValidationResult string
+	}{
+		{
+			name:          "none_breakglass",
+			justification: "issues/12345",
+			isBreakglass:  false,
+			wantValidationResult: fmt.Sprintf(`
+----- Justifications -----
+explanation    issues/12345
+
+----- Claims -----
+aud    [dev.abcxyz.jvs]
+exp    %s
+iat    %s
```

Comment:
We should be able to use https://github.com/abcxyz/jvs/blob/main/pkg/cli/token_create.go#L60 to set a fixed "now"? With that, can we turn the check to be string comparison (just compare the output string rather than parsing them)?

Reply:
The reason for needing parsing is fields like `jti (jwt id)` that in the validation result is not known before creating the token, so we have to parse the field and ignore those fields.

Updated to use `--now` to make test more stable.

Reply:
I see. Would using `-format=json` flag make it easier to parse? https://github.com/abcxyz/jvs/blob/main/pkg/cli/token_validate.go#L86

Reply:
I have considered that, but I think this will require creating some strict and unmarshel the json, and then doing diff with ignoreField. I think this is more complicated then the current method. Or maybe I am missing something here?

Reply:
I think being strict is what we want here. Otherwise we'd be introducing flakiness in the test. If we change the output format a little bit, the test will break.

Reply:
I have changed to use `-format=json` and `cmp.Diff` to do the compare.

After talking to Suhong, `--now` can only apply to breakglass. And if we use `format=json`, the timestamp will be in `RFC3339`, so I updated the timestamp comparison to not compare the seconds to make tests less flaky.